### DISCUSSION TITLE
Merging Selectors

### DISCUSSION THREAD

**Discussion by doolez** [jakubpawlowicz/clean-css#1226] [Q&A]
Hi everyone, I am hoping someone can help me with the following problem. I have tried multiple options for clean-css and can't seem to get it to merge all the properties for the selector with the end result being what I show under Desired Result. Is there a way to achieve this with clean-css?

### Desired Result
This is what I want: all properties merged for the same selector.
```
.banner {
  color: #fff;
  border-color: #fff
  display: flex;
  justify-content: center;
  flex-direction: column;
  height: 100%
  overflow: hidden;
  position: relative;
  clear: both
}
```

### Result after cleanCSS (level 0, rebase: false, format: 'beautify')
I think the key to the problem are the map source paths to why it's not working, but it feels like there should be a way to ignore them if that is the case.
```
/* line 30, ../../theme1/sass/placeholder-selectors/_colors.scss */
.banner {
  color: #fff;
  border-color: #fff
}

/* line 21, ../../theme1/sass/placeholder-selectors/_helpers.scss */
.banner {
  display: flex;
  justify-content: center;
  flex-direction: column;
  height: 100%
}

/* line 5, ../../theme2/sass/modules/_banner.scss */
.banner {
  overflow: hidden;
  position: relative;
  clear: both
}
```

### Result after cleanCSS (defaults)
The results are minified, but for legibility, I am placing them on separate lines below.
```
.banner{color:#fff;border-color:#fff}
.banner{display:flex;justify-content:center;flex-direction:column;height:100%}
.banner{overflow:hidden;position:relative;clear:both}
```

### Result after cleanCSS with options
```
{
  level: {
    2: {
      mergeSemantically: true,
      restructureRules: true
    }
  },
  compatibility: {
    properties: {
      merging: true
    },
    selectors: {
      mergeLimit: 8191
    }
  },
  rebase: false
}
```
Same as above, I am placing each one on separate lines below. These options combine the first two but not the last one. Are there any options that will combine them all?
```
.banner{color:#fff;border-color:#fff;display:flex;justify-content:center;flex-direction:column;height:100%}
.banner{overflow:hidden;position:relative;clear:both}
```

**Comment by jakubpawlowicz** [comment]
Hey @doolez - there are two possibilities here 

1) There's something between second and third `.banner` appearance which prevents merging those two. It can be another selector with same specificity that re-defines one of those `.banner{color:#fff;border-color:#fff;display:flex;justify-content:center;flex-direction:column;height:100%}` properties. Clean-css cannot reorder in such cases because it doesn't know the underlying HTML structure, so decides to play safe.

2) We explicitly don't do reordering with `flex` properties as it's a complex case.

Hope it helps somehow!
